tidy3d/components/data/data_array.py

```diff
@@ -129,39 +128,13 @@ def _interp_validator(self, field_name: str = None) -> None:
         if field_name is None:
             field_name = "DataArray"
 
-        dims = self.coords.dims
-
-        for dim in dims:
-            # in case we encounter some /0 or /NaN we'll ignore the warnings here
-            with np.errstate(divide="ignore", invalid="ignore"):
-                # check that we can interpolate
-                try:
-                    x0 = np.array(self.coords[dim][0])
-                    self.interp({dim: x0}, method="linear")
-                    self.interp({dim: x0}, method="nearest")
-                    # self.interp_like(self.isel({self.dim: 0}))
-                except pandas.errors.InvalidIndexError as e:
-                    raise DataError(
-                        f"'{field_name}.interp()' fails to interpolate along {dim} which is used by the solver. "
-                        "This may be caused, for instance, by duplicated data "
-                        f"in this dimension (you can verify this by running "
-                        f"'{field_name}={field_name}.drop_duplicates(dim=\"{dim}\")' "
-                        f"and interpolate with the new '{field_name}'). "
-                        "Plase make sure data can be interpolated."
-                    ) from e
-                # in case it can interpolate, try also to sel
-                try:
-                    x0 = np.array(self.coords[dim][0])
-                    self.sel({dim: x0}, method="nearest")
-                except pandas.errors.InvalidIndexError as e:
-                    raise DataError(
-                        f"'{field_name}.sel()' fails to select along {dim} which is used by the solver. "
-                        "This may be caused, for instance, by duplicated data "
-                        f"in this dimension (you can verify this by running "
-                        f"'{field_name}={field_name}.drop_duplicates(dim=\"{dim}\")' "
-                        f"and run 'sel()' with the new '{field_name}'). "
-                        "Plase make sure 'sel()' can be used on the 'DataArray'."
-                    ) from e
+        for dim, coord in self.coords.items():
+            if coord.to_index().duplicated().any():
+                raise DataError(
+                    f"Field '{field_name}' contains duplicate coordinates in dimension '{dim}'. "
+                    "You can remove duplicates by running "
+                    f"'{field_name}={field_name}.drop_duplicates(dim=\"{dim}\")'."
```

Comment:
Ah yeah good point, changed it.

Reply:
Yeah, sorry, this feels a bit odd to me so I have commented on it in the past, but I'm not 100% set against it and it may be hard to make sure we always enforce it...